auto-configurations/mcp/spring-ai-autoconfigure-mcp-server-common/src/main/java/org/springframework/ai/mcp/server/common/autoconfigure/McpServerAutoConfiguration.java

```diff
@@ -91,12 +96,42 @@ public class McpServerAutoConfiguration {
 
 	private static final LogAccessor logger = new LogAccessor(McpServerAutoConfiguration.class);
 
+	/**
+	 * Creates a configured ObjectMapper for MCP server JSON serialization.
+	 * <p>
+	 * This ObjectMapper is specifically configured for MCP protocol compliance with:
+	 * <ul>
+	 * <li>Lenient deserialization that doesn't fail on unknown properties</li>
+	 * <li>Proper handling of empty beans during serialization</li>
+	 * <li>Exclusion of null values from JSON output</li>
+	 * <li>Standard Jackson modules for Java 8, JSR-310, and Kotlin support</li>
+	 * </ul>
+	 * <p>
+	 * This bean can be overridden by providing a custom ObjectMapper bean with the name
+	 * "mcpServerObjectMapper".
+	 * @return configured ObjectMapper instance for MCP server operations
+	 */
+	@Bean(name = "mcpServerObjectMapper")
+	@ConditionalOnMissingBean(name = "mcpServerObjectMapper")
+	public ObjectMapper mcpServerObjectMapper() {
+		return JsonMapper.builder()
+			// Deserialization configuration
```

Comment:
I haven't looked in detail into this, but is there a way to remove the duplication between the two places where this PR configures an ObjectMapper (ie have one delegate to the other)?